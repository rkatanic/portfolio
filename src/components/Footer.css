.footer {
  display: flex;
  flex-direction: column;
  position: relative;
  text-align: center;
  color: var(--color-title);
  background-color: var(--color-primary-two);
}

.footer a,
.footer h3,
.footer p {
  color: hsla(0, 0%, 100%, 0.863);
}

.footer a:hover,
.footer-links svg:hover {
  fill: var(--color-input);
  color: var(--color-input);
}

.footer .nav-list {
  justify-content: center;
}

.footer-back-to-top {
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: var(--color-primary-two);
  width: 4rem;
  height: 4rem;
  border-radius: 50%;
  position: absolute;
  left: 50%;
  top: -1.5rem;
  transform: translateX(-50%);
}

.footer .icon-arrow-up {
  height: 70%;
  fill: hsla(0, 0%, 100%, 0.863);
  transition: 0.4s;
}

.footer .icon-arrow-up:hover {
  transform: translateY(-6px);
}

.footer-links {
  margin: 4rem 0 1rem;
  text-align: center;
}

.footer-links svg {
  height: 2rem;
  margin: 0 0.5rem;
  fill: hsla(0, 0%, 100%, 0.863);
}

.footer p {
  padding: 0.5rem 0;
  font-size: 0.75rem;
}

.footer-divider {
  position: relative;
}

hr {
  border: 0;
  height: 1px;
  width: 100%;
  position: relative;
  background-image: linear-gradient(
    to right,
    var(--color-primary-two),
    #fff,
    var(--color-primary-two)
  );
  background-color: #fff;
}

.center-diamond {
  position: absolute;
  width: 6px;
  height: 6px;
  background: #fff;
  top: 5px;
  left: 50%;
  transform: translate(-50%) rotate(-45deg);
}
