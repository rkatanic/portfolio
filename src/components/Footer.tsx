import { ReactComponent as ArrowUpIcon } from "../assets/icons/arrow-up.svg";
import { ReactComponent as GithubIcon } from "../assets/icons/github.svg";
import { ReactComponent as LinkedinIcon } from "../assets/icons/linkedin.svg";

import "./Footer.css";

const Footer = (): JSX.Element => (
  <div className="footer">
    <a href="#home" className="footer-back-to-top">
      <ArrowUpIcon />
    </a>
    <div className="footer-links">
      <a href="https://github.com/k-rade">
        <GithubIcon />
      </a>
      <a href="https://www.linkedin.com/in/radivoje-katanic">
        <LinkedinIcon />
      </a>
    </div>
    <ul className="nav-list">
      <li className="nav-item">
        <a href="#home" className="nav-link ">
          Home
        </a>
      </li>
      <li className="nav-item">
        <a href="#about" className="nav-link">
          About
        </a>
      </li>
      <li className="nav-item">
        <a href="#contact" className="nav-link">
          Contact
        </a>
      </li>
    </ul>
    <div className="footer-divider">
      <hr />
      <div className="center-diamond"></div>
    </div>
    <p>Created by:&nbsp; Radivoje Katanic</p>
  </div>
);

export default Footer;
