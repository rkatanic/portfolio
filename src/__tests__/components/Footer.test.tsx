import { render } from "@testing-library/react";
import Footer from "../../components/Footer";

describe("Footer", (): void => {
  it("should render", (): void => {
    const { baseElement } = render(<Footer />);

    expect(baseElement).toMatchSnapshot();
  });

  it("should redirect to correct github link", (): void => {
    const { getByText } = render(<Footer />);

    expect(getByText("GitHub")).toHaveAttribute(
      "href",
      "https://github.com/k-rade"
    );
  });

  it("should redirect to correct linkedin link", (): void => {
    const { getByText } = render(<Footer />);

    expect(getByText("LinkedIn")).toHaveAttribute(
      "href",
      "https://www.linkedin.com/in/radivoje-katanic"
    );
  });
});
