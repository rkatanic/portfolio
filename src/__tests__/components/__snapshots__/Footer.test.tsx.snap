// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render 1`] = `
<body>
  <div>
    <div
      class="footer"
    >
      <div
        class="footer-content"
      >
        <img
          alt=""
          class="footer-image"
          src="me1.png"
        />
        <div
          class="footer-links"
        >
          <a
            class="footer-link"
            href="https://www.linkedin.com/in/radivoje-katanic"
            target="_blank"
          >
            <div
              class="footer-link-icon"
            >
              <svg>
                linkedin.svg
              </svg>
            </div>
            LinkedIn
          </a>
          <a
            class="footer-link"
            href="https://github.com/k-rade"
            target="_blank"
          >
            <div
              class="footer-link-icon"
            >
              <svg>
                github.svg
              </svg>
            </div>
            GitHub
          </a>
        </div>
      </div>
      <div
        class="footer-actions"
      >
        <div
          class="color-switch"
        >
          <div
            class="color-switch-header"
          >
            <svg>
              pallete.svg
            </svg>
          </div>
        </div>
        <div
          class="night-mode-switch"
          tabindex="0"
        >
          <svg>
            sun.svg
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ReactModalPortal"
  />
</body>
`;
